src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,70 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int MAX_ARRAY_SIZE = Integer.MAX_VALUE;
+    private static final int DEFAULT_MULTIPLIER = 2;
```

Comment:
The `DEFAULT_MULTIPLIER` is declared as a static variable, which means it is shared across all instances of `MateStack`. Consider making it a non-static instance variable and marking it as final to ensure it remains constant after initialization. This will allow each instance of `MateStack` to have its own growth factor.